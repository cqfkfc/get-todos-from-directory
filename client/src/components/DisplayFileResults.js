import { List, ListItem, ListItemIcon, Typography } from "@material-ui/core";
import { Folder } from "@material-ui/icons";

// TODO: change the function name
function DisplayFileResults(props) {
  const { inputFiles, outputFiles } = props;

  return (
    <div data-testid="results">
      <Typography data-testid="msg-to-users" variant="h5">
        Out of the {inputFiles.length ? inputFiles.length : "no"} file
        {inputFiles.length === 1 ? "" : "s"} you uploaded,{" "}
        {outputFiles.length ? outputFiles.length : "no"} file
        {outputFiles.length === 1 ? "" : "s"} contained TODOs.
      </Typography>
      <List>
        {props.outputFiles.map((outputFile) => (
          <ListItem key={outputFile}>
            <ListItemIcon>
              <Folder />
            </ListItemIcon>
            {outputFile}
          </ListItem>
        ))}
      </List>
    </div>
  );
}

export default DisplayFileResults;
