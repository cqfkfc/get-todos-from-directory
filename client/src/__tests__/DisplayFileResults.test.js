import renderer from 'react-test-renderer';
import DisplayFileResults from "../components/DisplayFileResults"

describe('todo results show correctly', () => {
    test('when there are no files', ()=>{
        const dummyFiles = []
        const tree = renderer
            .create(<DisplayFileResults outputFiles={dummyFiles} inputFiles={dummyFiles}/>)
            .toJSON();
            expect(tree).toMatchSnapshot()
    })

    test('when there are 2 files', ()=>{
        const dummyFiles = ['PositiveFile1.js','PositiveFile2.js']
        const tree = renderer
            .create(<DisplayFileResults outputFiles={dummyFiles} inputFiles={dummyFiles}/>)
            .toJSON();
            expect(tree).toMatchSnapshot()
    })
})
