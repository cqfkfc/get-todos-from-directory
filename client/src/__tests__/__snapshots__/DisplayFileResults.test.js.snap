// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todo results show correctly when there are 2 files 1`] = `
<div
  data-testid="results"
>
  <h6
    className="MuiTypography-root MuiTypography-subtitle1"
    data-testid="msg-to-users"
  >
    Out of the 
    2
     file
    s
     you uploaded, 
    2
     file
    s
     contained TODOS.
  </h6>
  <ul
    className="MuiList-root MuiList-padding"
  >
    <li
      className="MuiListItem-root MuiListItem-gutters"
      disabled={false}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
          />
        </svg>
      </div>
      PositiveFile1.js
    </li>
    <li
      className="MuiListItem-root MuiListItem-gutters"
      disabled={false}
    >
      <div
        className="MuiListItemIcon-root"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
          />
        </svg>
      </div>
      PositiveFile2.js
    </li>
  </ul>
</div>
`;

exports[`todo results show correctly when there are no files 1`] = `
<div
  data-testid="results"
>
  <h6
    className="MuiTypography-root MuiTypography-subtitle1"
    data-testid="msg-to-users"
  >
    Out of the 
    0
     file
    
     you uploaded, 
    0
     file
    
     contained TODOS.
  </h6>
  <ul
    className="MuiList-root MuiList-padding"
  />
</div>
`;
